File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/jswidgets/formfield/DisabledStyleLookupCall.js
/*******************************************************************************
 * Copyright (c) 2017 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 ******************************************************************************/
jswidgets.DisabledStyleLookupCall = function() {
  jswidgets.DisabledStyleLookupCall.parent.call(this);
};
scout.inherits(jswidgets.DisabledStyleLookupCall, scout.StaticLookupCall);

jswidgets.DisabledStyleLookupCall.prototype._data = function() {
  return jswidgets.DisabledStyleLookupCall.DATA;
};

jswidgets.DisabledStyleLookupCall.DATA = [
  ['default', scout.Widget.DisabledStyle.DEFAULT],
  ['read only', scout.Widget.DisabledStyle.READ_ONLY]
];

File: modules/common/examples/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/ui/template/formfield/DisabledStyleLookupCall.java
/*
 * Copyright (c) 2021 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
package org.eclipse.scout.widgets.client.ui.template.formfield;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.scout.rt.client.ui.form.fields.IFormField;
import org.eclipse.scout.rt.platform.classid.ClassId;
import org.eclipse.scout.rt.shared.services.lookup.ILookupRow;
import org.eclipse.scout.rt.shared.services.lookup.LocalLookupCall;
import org.eclipse.scout.rt.shared.services.lookup.LookupRow;

@ClassId("1c303f17-674e-4f99-b94e-00ea4db42df8")
public class DisabledStyleLookupCall extends LocalLookupCall<Integer> {

  private static final long serialVersionUID = 1L;

  @Override
  protected List<? extends ILookupRow<Integer>> execCreateLookupRows() {
    List<ILookupRow<Integer>> rows = new ArrayList<>();
    rows.add(new LookupRow<>(IFormField.DISABLED_STYLE_DEFAULT, "default"));
    rows.add(new LookupRow<>(IFormField.DISABLED_STYLE_READ_ONLY, "read only"));
    return rows;
  }
}
